x-pack/solutions/observability/plugins/apm/public/components/shared/environment_select/index.tsx

```diff
@@ -53,60 +68,80 @@ export function EnvironmentSelect({
   onChange: (value: string) => void;
 }) {
   const [searchValue, setSearchValue] = useState('');
-
-  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
-
-  const selectedOptions: Array<EuiComboBoxOptionOption<string>> = [
+  const [selectedOption, setSelectedOption] = useState<Array<EuiComboBoxOptionOption<string>>>([
     {
       value: environment,
       label: getEnvironmentLabel(environment),
     },
-  ];
+  ]);
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+  const { data, onSearchChange, searchStatus } = useEnvironmentSelect({ serviceName, start, end });
 
   const onSelect = (changedOptions: Array<EuiComboBoxOptionOption<string>>) => {
     if (changedOptions.length === 1 && changedOptions[0].value) {
       onChange(changedOptions[0].value);
     }
+    setSelectedOption(changedOptions);
   };
 
-  const { data, status: searchStatus } = useFetcher(
-    (callApmApi) => {
-      return isEmpty(searchValue)
-        ? Promise.resolve({ terms: [] })
-        : callApmApi('GET /internal/apm/suggestions', {
-            params: {
-              query: {
-                fieldName: SERVICE_ENVIRONMENT,
-                fieldValue: searchValue,
-                serviceName,
-                start,
-                end,
-              },
-            },
-          });
+  const terms = useMemo(() => {
+    if (isEmpty(searchValue)) {
+      return availableEnvironments;
+    }
+
+    return [...new Set(data?.terms.concat(...availableEnvironments))];
+  }, [availableEnvironments, data?.terms, searchValue]);
+
+  const options = useMemo<Array<EuiComboBoxOptionOption<string>>>(() => {
+    const environmentOptions = getEnvironmentOptions(terms);
+
+    return isEmpty(searchValue)
```

Comment:
Nit I saw that in `isInvalid` on line 106 we use `searchValue !== ''` can we do the same or just check `searchValue` and remove the `isEmpty` - also in the other places and we will have it consistent in the file + reduce the dependencies 😎 
```suggestion
    return searchValue !== ''
```